File: VB/WindowsApplication1/CustomEditor/ExtendedButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator

Namespace WindowsApplication1
  <System.ComponentModel.DesignerCategory("")> _
  Public Class ExtendedButtonEdit
	  Inherits ButtonEdit
		Shared Sub New()
			RepositoryItemExtendedButtonEdit.Register()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemExtendedButtonEdit.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemExtendedButtonEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemExtendedButtonEdit)
			End Get
		End Property
  End Class
End Namespace

File: CS/CS/WindowsApplication1/CustomEditor/ExtendedButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;

namespace WindowsApplication1
{
  [System.ComponentModel.DesignerCategory("")]
    public class ExtendedButtonEdit : ButtonEdit
    {
        static ExtendedButtonEdit()
        {
            RepositoryItemExtendedButtonEdit.Register();
        }
     
        public override string EditorTypeName
        {
            get { return RepositoryItemExtendedButtonEdit.EditorName; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemExtendedButtonEdit Properties
        {
            get { return base.Properties as RepositoryItemExtendedButtonEdit; }
        }
    }
}